### PR TITLE
Add Travis-CI deploy action for Javadocs

### PR DESCRIPTION
**Pull request by fractalwrench**
This creates a Travis-CI deploy action that generates Javadocs and uploads them to Github Pages, whenever a change is pushed to the master branch.

### CODE REVIEW

**Comment by kattrali** [comment]
Hmm. I don't think we should check the docs in, instead we can run the `javadoc` command in a `before_deploy` step and deploy the docs dir contents generated by that action. It might get a bit noisy otherwise.

**Comment by fractalwrench** [comment]
I've added `docs/` to the gitignore and updated the travis YML accordingly.

File: .travis.yml [hunk 1]
```diff
@@ -32,3 +32,13 @@ before_script:
...
+
+before_deploy: ./gradlew javadoc
+
+deploy:
+  provider: pages
+  local_dir: docs # only include the contents of the docs dir
+  skip_cleanup: true
+  github_token: $GITHUB_TOKEN # Set in travis-ci.org dashboard
+  on:
+    branch: master # deploy when changes pushed to this branch
```
Comment (kattrali) [inline .travis.yml]: This should be on tags instead of master to ensure only released content is being documented

Comment (fractalwrench) [inline .travis.yml]: This will now only deploy Javadoc when a tag is applied to a commit.

**Pull request merged**